content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc

```diff
@@ -299,99 +449,180 @@ include::example$code-samples.cs[tags=ListenerValidationTestMode, indent=0]
+
+. Enter `SeasonalItemsPage` by selecting "What's in Season?" from the "hamburger" menu.
+
+. Edit the quantity and/or image on one or multiple instance(s) and press Save when you are done editing
+
+. Watch it sync automatically to other connected clients
+
+****
 
 .App in action -- sync
-====
-image::xamarin-sync.gif[alt=server websocket listener login screen,width=600,height=320]
-====
+image::xamarin-sync.gif[alt=server websocket listener login screen]
+
 
 == What Next
-As an exercise, switch between the various TLS modes and server cert validation modes and see how the app behaves. You can also try with different topologies to connect the peers.
+
+
+As an exercise, switch between the various TLS modes and server cert validation modes and see how the app behaves.
+You can also try with different topologies to connect the peers.
+
 
 == Learn More
-Congratulations on completing this tutorial!
 
-This tutorial walked you through an example of how to directly synchronize data between Couchbase Lite enabled clients. While the tutorial is for Xamarin .Net, the concepts apply equally to other Couchbase Lite platforms.
 
+Congratulations on completing this tutorial!
+
+This tutorial walked you through an example of how to directly synchronize data between Couchbase Lite enabled clients.
+While the tutorial is for iOS, the concepts apply equally to other Couchbase Lite platforms.
 
-=== Further Reading
+.Further Reading
+****
+Complete documentation is available
+xref:couchbase-lite:{module}:p2psync-websocket.adoc[here]
+****
 
-Complete documentation is available link:https://docs.couchbase.com/couchbase-lite/current/csharp/learn/csharp-landing-p2psync.html[here]
 
 == Troubleshoot
 
+[discrete]
 === Having issue running the Xamarin iOS app?
+
 * Xamarin iOS p2p sample app should build and run with Visual Studio 2019 with latest updates and XCode 12.0.1
-* If you have Xcode 11 and try to run the iOS app on simulator and the simulator is not loading, try launch the simulator via Xcode and select that simulator when launching it from VS.
+* If you have Xcode 11 and try to run the iOS app on simulator and the simulator is not loading, try to launch the simulator via Xcode and select that simulator when launching it from VS.
+
+[discrete]
+=== Connecting to an Android emulator
 
-=== Connecting to Android emulator over localhost
-When starting a listener on Android emulator and trying to connect it from another emulator or iOS simulator on localhost, the following steps must be followed
+IMPORTANT: Sync will not work between two emulators. At least one app must be running on a device.
 
-. Setup port forwarding. +
+When starting a listener on Android emulator and trying to connect it from a device or iOS simulator on localhost, the following steps must be followed:
+
+. Hard-code a port so you know what EP to use
```

Comment:
@ibsoln  Is "EP" supposed to be endpoint. If so, pls. expand. Its not an acronym.  Applies globally

Reply:
> 

It is. I'll fix it.